README.md

```diff
@@ -115,6 +115,62 @@ Server
 In many rails projects when you run the server with `rails s` Rails sets RAILS_ENV to "development".  If you do that with this app, you will connect to the development database on AWS (if you have permission to decrypt the value).  Instead, run `RAILS_ENV=local rails s` to start the server and `RAILS_ENV=local rails c` to run the console.
 
 
+Asynchronous Execution
+========================
+Why: Our asynchronous code is currently used for user barcode creation on account create.  During that process, we need to talk to Sierra one or more times, and those calls can take up time and resources.  We'd like the user to be able to peruse the site while their barcode is being computed.  For more info on barcodes, see https://confluence.nypl.org/display/DIGTL/User+Barcodes .
+
+Our asynchronous functionality is done through ActiveJob, which employs DelayedJob on the backend.
+ActiveJob gets better after rails 5.2, so keep in mind that there is some functionality that is not perfect, until we can upgrade rails.
+
+Backstory:  Rails runs on a single thread within a single process.  When you send a request with HTTParty, that request is executed on the same thread as the rest of the application.  Rails will sometimes simulate asynchronicity with creative use of scheduling on IO operations on some of the Rails servers/environments.  But, generally, HttParty will be running on the same thread and process as your app, and will hold up app execution, if it's stuck.
+
+To make code asynchronous, we're making a second worker, which will start its own process, which will spin its own thread:
```

Comment:
I don't think it's accurate that a process will "spin its own thread". If there's only one thread then the thread is the process. 
I think it's clearer to just say "To make code asynchronous, we're making a second worker" and leave it at that.

Reply:
@daryanypl, as mentioned in stand-up, if you have a reference for the "worker starts a process which spins up a thread" please link it. It seems like you are drawing a distinction between a worker, a process and a [single] thread. That may well be correct — I'd be hapy to be wrong and learn something — but I don't understand the distinction is.

Reply:
One place is here:
https://github.com/collectiveidea/delayed_job
"# Runs two workers in separate processes.
RAILS_ENV=production script/delayed_job -n 2 start"

This (https://medium.com/@adrianbooth/exploring-rubys-multi-threaded-weirdness-d5dc61883591) is not a direct answer, but I found it helpful in learning about RoR quirks, and in interpreting all of the other information I've been reading on this topic.  It has visual diagrams, as well as verbal explanations, and I'm a visual learner.

Reply:
Hi Darya, as I said above (and in standup) the thing I think is wrong is to say the process will "spin its own thread". Certainly a "worker" can be a process; a worker in a different implementation could also be a thread within a process. Those links are very interesting, and good for dicussion, but I don't see the thing I was objecting to addressed in either of them.